### PR TITLE
issue-414 : fixed link creation

### LINKED ISSUE #414: links in ES shall be updated
In application-data where we store link information in Elastic search, we need to check if links are configured.

Note: few links are available, but has only SOURCE. It also needs SINK for running testcases

### PR DESCRIPTION
**Pull request by DanaSunal**
Fixes #414

### CODE REVIEW

File: server/service/individualServices/LinkServices.js [hunk 1]
```diff
@@ -204,24 +208,26 @@ function isConsumingServiceExists(link, consumingOperationUuid) {
...
+            [onfAttributes.LINK.LOGICAL_TERMINATION_POINT] : consumingOperationUuid
+        };
+        link[onfAttributes.LINK.LINK_PORT].push(consumingOperationLinkPort);
+    }
     let servingOperationLocalId = LinkPort.generateNextLocalId(link);
     let servingOperationLinkPort = {
         [onfAttributes.LOCAL_CLASS.LOCAL_ID] : servingOperationLocalId,
-        [onfAttributes.LINK.PORT_DIRECTION] : LinkPort.portDirectionEnum.OUTPUT,
-        [onfAttributes.LINK.LOGICAL_TERMINATION_POINT] : servingOperationuuid
+        [onfAttributes.LINK.PORT_DIRECTION] : LinkPort.portDirectionEnum.INPUT,
```
Comment (PrathibaJee) [inline server/service/individualServices/LinkServices.js:229]: Kindly let me know is there any reason to change the PortDirection from OUTPUT to INPUT ?

File: server/service/individualServices/LinkServices.js [hunk 2]
```diff
@@ -38,8 +39,8 @@ exports.findOrCreateLinkForTheEndPointsAsync = async function (EndPoints) {
     took += consumingEndpointResponse.took;
     let consumingOperationUuid = consumingEndpointResponse.consumingOperationUuid;
 
-    if (servingOperationUuid && consumingOperationUuid) {
-        let linkResponse = await getLinkOfTheOperationAsync(servingOperationUuid, LinkPort.portDirectionEnum.OUTPUT);
+    if (servingOperationUuid) {
+        let linkResponse = await getLinkOfTheOperationAsync(servingOperationUuid, LinkPort.portDirectionEnum.INPUT);
```
Comment (PrathibaJee) [inline server/service/individualServices/LinkServices.js:43]: Kindly let me know is there any reason to change the PortDirection from OUTPUT to INPUT ?

**Review by PrathibaJee — CHANGES_REQUESTED** [review]

File: server/service/individualServices/LinkServices.js (hunk 1 above)
Comment (DanaSunal) [reply server/service/individualServices/LinkServices.js:229]: Yes, I think it was wrong in the first place. If my understanding is correct, serving applications have direction Source ( LinkPort.portDirectionEnum.INPUT ) and consuming applications have direction Sink ( LinkPort.portDirectionEnum.OUTPUT).

File: server/service/individualServices/LinkServices.js (hunk 2 above)
Comment (DanaSunal) [reply server/service/individualServices/LinkServices.js:43]: Same as above.

File: server/service/individualServices/LinkServices.js (hunk 1 above)
Comment (PrathibaJee) [reply server/service/individualServices/LinkServices.js:229]: Yes Serving application is Source, and consuming application is Sink.
As per the description mentioned in the specification for the /v1/list-links-to-operation-clients-of-application , operationServer = SOURCE= OUTPUT , operationClient = SINK= INPUT
image

**Review by PrathibaJee — APPROVED** [review]

**Pull request merged**